echo-library/src/common/starlark_modules.rs

```diff
@@ -1,5 +1,83 @@
 use super::*;
 use anyhow::anyhow;
+
+fn validate_value(value: &serde_json::Value, expected_type: &RustType) -> anyhow::Result<()> {
+    match expected_type {
+        RustType::String => {
+            if !value.is_string() {
+                return Err(anyhow!("Value must be a string"));
+            }
+        }
+        RustType::Int => {
+            if !value.is_i64() {
+                return Err(anyhow!("Value must be an integer"));
+            }
+        }
+        RustType::Float => {
+            if !value.is_f64() {
+                return Err(anyhow!("Value must be a float"));
+            }
+        }
+        RustType::Uint => {
+            if !value.is_u64() {
+                return Err(anyhow!("Value must be a positive integer"));
+            }
+        }
+        RustType::Boolean => {
+            if !value.is_boolean() {
+                return Err(anyhow!("Value must be a boolean"));
+            }
+        }
+
+        RustType::HashMap(key_type, value_type) => {
+            if let Some(map) = value.as_object() {
+                for (key, val) in map.iter() {
+                    validate_value(val, value_type)?;
+                }
+            } else {
+                return Err(anyhow!("Value must be a JSON object"));
+            }
+        }
+
+        RustType::List(item_type) => {
+            let parsed_array = value
+                .as_array()
+                .ok_or_else(|| anyhow!("Value must be a JSON array"))?;
+
+            for element in parsed_array.iter() {
+                validate_value(element, item_type)?;
+            }
+        }
+
+        RustType::Tuple(key_type, value_type) => {
+            let parsed_tuple = value
+                .as_array()
+                .ok_or_else(|| anyhow!("Value must be a JSON tuple with two elements"))?;
+
+            if parsed_tuple.len() != 2 {
+                return Err(anyhow!("Tuple must have exactly two elements"));
+            }
+
+            // Pattern matching for key and value types
+            match (&parsed_tuple[0], &parsed_tuple[1]) {
+                (serde_json::Value::String(key), value) => validate_value(value, value_type)?,
+                (serde_json::Value::Number(number), value) if number.is_i64() => {
+                    validate_value(value, value_type)?
+                }
+                (serde_json::Value::Number(number), value) if number.is_f64() => {
+                    validate_value(value, value_type)?
+                }
+                (serde_json::Value::Bool(bool_value), value) => validate_value(value, value_type)?,
```

Comment:
Validation for tuple types, hashmaps, arrays, and similar structures is not handled.